massa-protocol-worker/src/worker_operations_impl.rs

```diff
@@ -23,13 +23,53 @@ use tokio::time::{sleep_until, Instant, Sleep};
 use tracing::warn;
 
 impl ProtocolWorker {
+    /// Limit the number of batch an external node can send in a second. Ban
+    /// it if it exceed self.settings.max_op_batch_per_sec_per_node
+    ///
+    /// Call that function each time we receive a batch from a `node_id`
+    ///
+    /// Return true if node has been considered as bannished, false otherwise.
+    async fn limitation_batches_by_node(&mut self, node_id: NodeId) -> bool {
```

Comment:
There is something odd in my opinion with that function name maybe batches by node limitation or limit batches by node